blast/formatter/server/internal/server/run-report-ep.go

```diff
@@ -102,16 +103,41 @@ func runReport(job *api.JobPayload, log *logrus.Entry) midl.Response {
 }
 
 // zipDir zips the contents of the given directory.
-func zipDir(path string) error {
-	if matches, err := filepath.Glob(filepath.Join(path, "*")); err != nil {
-		return err
-	} else {
-		if err := xfiles.ZipFiles(path, reportExportName, matches); err != nil {
+func zipDir(path string) (err error) {
+	var matches []string
+
+	// Loop to wait for the report file if we get here before it's copied into the
+	// workspace directory.
+	//
+	// Try a maximum of 10 times before bailing on the job (something is wrong if
+	// we've waited a whole second and the report isn't here yet).
+	for i := 0; i < 10; i++ {
+
+		// Search for files in the workspace directory (will be empty until the
```

Comment:
Maybe a longer comment here about how you know when it's ok to zip.  Looks like you are zipping a whole dir but you are zipping as soon as you see "the report"; is that the "main" result file, so it's ok?  Should we wait until the .completed flag file?  Some more explanation might help.

Reply:
Meant to put this on the comment on line 123

Reply:
The `meta.json` file and completed flag are written after the zip is created.  The only things going into the zip files are the report outputs